pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -128,19 +128,21 @@ public void initialize(URL location, ResourceBundle resources) {
 
 
 
-        EventStream<Integer> changesEventStream = xpathExpressionArea.plainTextChanges().map(m -> m.getInsertionEnd());
-
-        EventStream<KeyEvent> keyPressed = EventStreams.eventsOf(xpathExpressionArea, KeyEvent.KEY_PRESSED);
-
-        EventStream<Integer> keyCombo = keyPressed
-            .filter(key -> (key.getCode().equals(KeyCode.CONTROL) && key.getCode().equals(KeyCode.SPACE)))
+        EventStream<Integer> changesEventStream = xpathExpressionArea.plainTextChanges()
+                                                                     .map(m -> {
+                                                                         if (m.getRemoved().length() > 0) {
+                                                                             return m.getRemovalEnd();
+                                                                         }
+                                                                         return m.getInsertionEnd();
+                                                                     });
+
+        EventStream<Integer> keyCombo = EventStreams.eventsOf(xpathExpressionArea, KeyEvent.KEY_PRESSED)
+            .filter(key -> (key.isControlDown() && key.getCode().equals(KeyCode.SPACE)))
             .map(key -> xpathExpressionArea.getCaretPosition());
 
-        EventStream<Integer> keyEvents = EventStreams.merge(keyCombo, changesEventStream);
-        keyEvents.map(m -> {
-
-            int indexOfSlash = reverseStringSearch('/', xpathExpressionArea.getText(), m);
-            String input = xpathExpressionArea.getText().substring(indexOfSlash + 1, m);
+        EventStreams.merge(keyCombo, changesEventStream).map(searchPoint -> {
+            int indexOfSlash = xpathExpressionArea.getText().lastIndexOf(",", searchPoint - 1);
```

Comment:
Instead of using the index of the slash, we could use the index of the character right after to avoid having to add +1 (eg when replacing, which for now misses a character) each time.